abot.go

```diff
@@ -304,15 +312,15 @@ func installPlugins() {
 		if err.Error() == "exit status 1" {
 			l.Info("Is a plugin trying to import a non-existent package?")
 		}
-		l.Fatal("Failed to install plugins.", err)
+		return err
 	}
 
 	// Sync each of them to get dependencies
 	var wg sync.WaitGroup
 	wg.Add(len(plugins.Dependencies))
 	rand.Seed(time.Now().UTC().UnixNano())
 	for url, version := range plugins.Dependencies {
-		go func(url, version string) {
+		go func(url, version string) error {
```

Comment:
My understanding is that each goroutine will overwrite the returned `error`, so in this case we'll be left with whichever value the final goroutine set error equal to.

To get around this, we can use channels like this example (http://www.atatus.com/blog/goroutines-error-handling/). We can send the errors as messages, which can then each be handled as they come into the channel.